haystack/modeling/training/base.py

```diff
@@ -372,30 +372,27 @@ def train(self):
 
     def compute_loss(self, batch: dict, step: int) -> torch.Tensor:
         # Forward & backward pass through model
-        logits = self.model.forward(**batch)
-        per_sample_loss = self.model.logits_to_loss(logits=logits, global_step=self.global_step, **batch)
-        return self.backward_propagate(per_sample_loss, step)
+        with torch.cuda.amp.autocast(enabled=self.use_amp):
+            logits = self.model.forward(**batch)
+            per_sample_loss = self.model.logits_to_loss(logits=logits, global_step=self.global_step, **batch)
+            loss = self.adjust_loss(per_sample_loss)
+        return self.backward_propagate(loss, step)
 
     def backward_propagate(self, loss: torch.Tensor, step: int):
-        loss = self.adjust_loss(loss)
```

Comment:
I needed to move this out of `backward_propagate` because Pytorch shows that the `adjust_loss` step should happen within the same `autocast` context. Link to explanation: https://pytorch.org/docs/stable/notes/amp_examples.html#gradient-accumulation